cjdns-hdr/src/data_header.rs

```diff
@@ -0,0 +1,150 @@
+//! Logic for a simple data header, providing type of content
+
+use crate::{
+    errors::{HeaderError, Result},
+    utils::{Reader, Writer},
+};
+
+const DATA_HEADER_SIZE: usize = 4;
+const HEADER_CURRENT_VERSION: u8 = 1;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct DataHeader {
+    pub version: u8,
+    pub content_type: header_content::ContentType,
+}
+
+impl DataHeader {
+    /// Parses bytes into `DataHeader` struct.
+    ///
+    /// Results in error if bytes length doesn't equal to 4
+    pub fn parse(data: &[u8]) -> Result<Self> {
+        if data.len() != DATA_HEADER_SIZE {
+            return Err(HeaderError::CannotParse("invalid header data size"));
+        }
+        let mut data_reader = Reader::new(data);
+        let version = {
+            let version_with_flags = data_reader.read_u8().expect("invalid header data size");
+            version_with_flags >> 4
```

Comment:
How about flags? Do we need them or just throwing away?

Reply:
just throwing away